document-store/src/main/java/org/hypertrace/core/documentstore/expression/LiteralExpression.java

```diff
@@ -0,0 +1,17 @@
+package org.hypertrace.core.documentstore.expression;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+@EqualsAndHashCode
+public class LiteralExpression implements Expression {
```

Comment:
How is this different than a constant expression?

Reply:
Literal expressions are for variables like 
`SELECT MIN(col1) FROM table`
whereas constant expressions are for specifying constants like
`SELECT SUM(1) FROM table`

Specifically, Mongo requires literals to be prefixed with "$", whereas constants should be used without any prefix.

Reply:
Makes sense. I will rename LiteralExpression to IdentifierExpression in the new API PR (https://github.com/hypertrace/document-store/pull/58)